Improve reliability and debuggability of http/wpt/mediarecorder/pause-recording.html and http/wpt/mediarecorder/MediaRecorder-audio-bitrate-webm.html

youennf opened the pull request:
#### f2bf55e652d728e60cdc061123fe29cb31a277ec
<pre>
Improve reliability and debuggability of http/wpt/mediarecorder/pause-recording.html and http/wpt/mediarecorder/MediaRecorder-audio-bitrate-webm.html
<a href="https://rdar.apple.com/141616340">rdar://141616340</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=284816">https://bugs.webkit.org/show_bug.cgi?id=284816</a>

Reviewed by Jean-Yves Avenard.

We mark those tests as slow.
We make sure that for the audio bitrate test, we gather enough data to be able to compare correctly different bitrates.
We make sure to stop tracks in http/wpt/mediarecorder/pause-recording.html to improve efficiency.

We move one of the test in its own http/wpt/mediarecorder/pause-recording-2.html since it appears to be more flaky than other tests.
We update the test by fixing a typo in the name of a promise that is reused lowercase while it is defined mixedcase.
And we add a helper routine to wait for events with a timer so that the test will give more information in case it is timing out.

* LayoutTests/TestExpectations:
* LayoutTests/http/wpt/mediarecorder/MediaRecorder-audio-bitrate.js:
(async record):
(record): Deleted.
* LayoutTests/http/wpt/mediarecorder/pause-recording-2-expected.txt: Added.
* LayoutTests/http/wpt/mediarecorder/pause-recording-2.html: Added.
* LayoutTests/http/wpt/mediarecorder/pause-recording-expected.txt:
* LayoutTests/http/wpt/mediarecorder/pause-recording.html:

Canonical link: <a href="https://commits.webkit.org/287950@main">https://commits.webkit.org/287950@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c32230d2f294e0e594e64e036b1ea2c715c1dc41

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/81470 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/995 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35413 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/85999 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/32460 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1013 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/8814 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/63589 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/21332 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/84539 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/708 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74150 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/43882 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/604 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/28315 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/30914 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/72070 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/28903 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/87434 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/8700 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/6185 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/71904 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/8881 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/69971 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71142 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17705 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15206 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14119 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/8662 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/8498 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12019 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10306 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c32230d2f294e0e594e64e036b1ea2c715c1dc41. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/38075

jyavenard commented on LayoutTests/http/wpt/mediarecorder/pause-recording-2.html:
```diff
@@ -0,0 +1,66 @@
+    <script>
+function waitFor(duration)
+{
+    return new Promise((resolve) => setTimeout(resolve, duration));
```
those methods exists already in multiple location, can we reuse those instead?

jyavenard commented on LayoutTests/http/wpt/mediarecorder/pause-recording-2.html:
```diff
@@ -0,0 +1,66 @@
+function waitForEvent(object, eventName, testName)
+{
+    return new Promise((resolve, reject) => {
+        object.addEventListener(eventName, resolve, { once: true });
```
Same here, named waitFor(object, eventName)

youennf committed:
Improve reliability and debuggability of http/wpt/mediarecorder/pause-recording.html and http/wpt/mediarecorder/MediaRecorder-audio-bitrate-webm.html
rdar://141616340
https://bugs.webkit.org/show_bug.cgi?id=284816

Reviewed by Jean-Yves Avenard.

We mark those tests as slow.
We make sure that for the audio bitrate test, we gather enough data to be able to compare correctly different bitrates.
We make sure to stop tracks in http/wpt/mediarecorder/pause-recording.html to improve efficiency.

We move one of the test in its own http/wpt/mediarecorder/pause-recording-2.html since it appears to be more flaky than other tests.
We update the test by fixing a typo in the name of a promise that is reused lowercase while it is defined mixedcase.
And we add a helper routine to wait for events with a timer so that the test will give more information in case it is timing out.

* LayoutTests/TestExpectations:
* LayoutTests/http/wpt/mediarecorder/MediaRecorder-audio-bitrate.js:
(async record):
(record): Deleted.
* LayoutTests/http/wpt/mediarecorder/pause-recording-2-expected.txt: Added.
* LayoutTests/http/wpt/mediarecorder/pause-recording-2.html: Added.
* LayoutTests/http/wpt/mediarecorder/pause-recording-expected.txt:
* LayoutTests/http/wpt/mediarecorder/pause-recording.html:

Canonical link: https://commits.webkit.org/287950@main
